app/src/main/java/com/github/featuredetectandroid/ui/OutputViewModel.kt

```diff
@@ -0,0 +1,33 @@
+package com.github.featuredetectandroid.ui
+
+import android.graphics.Bitmap
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.geometry.Offset
+import androidx.lifecycle.ViewModel
+import com.github.featuredetectandroid.utils.conversions.keypointsToOffsetList
+import com.github.featuredetectandroid.utils.conversions.luminanceArrayToBitmap
+import com.github.featuredetectlib.FeatureDetector
+import com.github.featuredetectlib.Keypoint
+
+class OutputViewModel : ViewModel() {
+    var grayscaleBitmap: Bitmap? by mutableStateOf(null)
+    var isCameraPermissionGranted by mutableStateOf(false)
+    var keypointsOffset: List<Offset> by mutableStateOf(emptyList())
+    var featureDetector: FeatureDetector? by mutableStateOf(null)
+    private var width by mutableStateOf(0)
+    private var height by mutableStateOf(0)
+
+    fun setPicture(grayscaleByteArray: ByteArray, newWidth: Int, newHeight: Int) {
+        width = newWidth
+        height = newHeight
+        grayscaleBitmap = luminanceArrayToBitmap(grayscaleByteArray, newWidth, newHeight)
+    }
+
+    fun setKeypointsForOutput(keypoints: List<Keypoint>) {
+        keypointsOffset = keypointsToOffsetList(keypoints)
+    }
+
+    fun getSize() = Pair(width, height)
```

Comment:
Its usage is pretty awkward. If you'll manage to set `width` and `height` only when they are really changed and not for each frame, it would be better to just make them `public` and set/get them from their setters/getters.

Reply:
fixed